src/com/goide/inspections/GoMixedNamedUnnamedParametersInspection.java

```diff
@@ -0,0 +1,93 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections;
+
+import com.goide.psi.*;
+import com.goide.runconfig.testing.GoTestFinder;
+import com.goide.runconfig.testing.GoTestFunctionType;
+import com.intellij.codeInspection.LocalInspectionToolSession;
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.progress.ProgressManager;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class GoMixedNamedUnnamedParametersInspection extends GoInspectionBase {
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+      @Override
+      public void visitMethodDeclaration(@NotNull GoMethodDeclaration o) {
+        super.visitMethodDeclaration(o);
+        visitDeclaration(holder, o.getSignature(), "Method");
+      }
+
+      @Override
+      public void visitFunctionDeclaration(@NotNull GoFunctionDeclaration o) {
+        super.visitFunctionDeclaration(o);
+        if (GoTestFinder.isTestFile(o.getContainingFile()) && GoTestFunctionType.fromName(o.getName()) != null) {
+          return;
+        }
+        visitDeclaration(holder, o.getSignature(), "Function");
+      }
+
+      @Override
+      public void visitFunctionLit(@NotNull GoFunctionLit o) {
+        super.visitFunctionLit(o);
+        visitDeclaration(holder, o.getSignature(), "Closure");
+      }
+    };
+  }
+
+  private static void visitDeclaration(@NotNull ProblemsHolder holder, @Nullable GoSignature signature, String ownerType) {
+    if (signature == null) return;
+    GoParameters parameters = signature.getParameters();
+    visitParameterList(holder, parameters, ownerType, "parameters");
+
+    GoResult result = signature.getResult();
+    parameters = result != null ? result.getParameters() : null;
+    visitParameterList(holder, parameters, ownerType, "return parameters");
+  }
+
+  private static void visitParameterList(
+    @NotNull ProblemsHolder holder,
+    @Nullable GoParameters parameters,
+    String ownerType,
+    String what) {
+
+    if (parameters == null || parameters.getParameterDeclarationList().isEmpty()) return;
+    ProgressManager.checkCanceled();
+    boolean hasNamed = false;
+    boolean hasUnnamed = false;
+    for (GoParameterDeclaration parameterDeclaration : parameters.getParameterDeclarationList()) {
+      if (parameterDeclaration.getParamDefinitionList().isEmpty()) {
+        hasUnnamed = true;
+      }
+      else {
+        hasNamed = true;
+      }
+    }
+
+    if (hasNamed && hasUnnamed) {
```

Comment:
What I've noticed while working on this was that there's a difference between the Go parser and the plugin parser for the following Parameters ` a, b int, string, int `:
- the plugin parser sees it as: ` a, b int // ParamDecl1, string // unamed ParamDecl2, int // unamed ParamDecl3 ` 
- the Go parser sees it as: ` a, b int // ParamDecl1, string, int // ParamDecl2 + missing type info `

I can move it in the for loop but it also depends on how the reporting should be done for this. Should all the parameter/return list be flagged?

The quick fix for this one is not immediately clear, maybe the quickfix should be postponed until the change signature capability lands (if ever done at all)?